src/clojure/nrepl/misc.clj

```diff
@@ -78,30 +78,13 @@
           (when log?
             (log e))))))
 
-(defmacro with-session-classloader
-  "This macro does two things:
-
-   1. If the session has a classloader set, then execute the body using that.
-      This is typically used to trigger the sideloader, when active.
-
-   2. Bind `clojure.lang.Compiler/LOADER` to the context classloader, which
-      might also be the sideloader. This is required to get hotloading with
-      pomegranate working under certain conditions."
-  [session & body]
-  `(let [ctxcl#  (.getContextClassLoader (Thread/currentThread))
-         alt-cl# (when-let [classloader# (:classloader (meta ~session))]
-                   (classloader#))
-         cl#     (or alt-cl# ctxcl#)]
-     (if (= ctxcl# cl#)
-       (with-bindings {clojure.lang.Compiler/LOADER cl#}
-         ~@body)
-       (do
-         (.setContextClassLoader (Thread/currentThread) cl#)
-         (try
-           (with-bindings {clojure.lang.Compiler/LOADER cl#}
-             ~@body)
-           (finally
-             (.setContextClassLoader (Thread/currentThread) ctxcl#)))))))
+(defmacro bind-classloader
```

Comment:
Why `bind-classloader` instead of `with-classloader`? I thought `with-*` names were more common when a macro has a body.

Reply:
That's true. I'm pretty sure this macro will also be gone in the future. But anyway, I'll rename it to keep the convention for now.